mesh/amqp.c

```diff
@@ -460,125 +593,143 @@ void mesh_amqp_start(struct mesh_amqp *amqp)
 void mesh_amqp_free(struct mesh_amqp *amqp)
 {
 	if (amqp->thread_started) {
-		mesh_amqp_close(amqp);
+		mesh_amqp_stop(amqp);
 		pthread_join(amqp->thread, NULL);
```

Comment:
I'd rather return thread context from thread function and free it here, just to keep deallocation in the same "place" as allocation.